api/src/main/java/org/apache/iceberg/expressions/BoundUnaryPredicate.java

```diff
@@ -46,18 +46,35 @@ public boolean test(T value) {
         return value == null;
       case NOT_NULL:
         return value != null;
+      case IS_NAN:
+        return isNaN(value);
+      case NOT_NAN:
+        return !isNaN(value);
       default:
         throw new IllegalStateException("Invalid operation for BoundUnaryPredicate: " + op());
     }
   }
 
+  private boolean isNaN(T value) {
```

Comment:
Should this be a static helper somewhere since it is duplicated here and in `Evaluator`?